### PR TITLE
make sumoperator and  PotentitalOperator hash stable / comparable

### PR DESCRIPTION
**Pull request by PhilipVinc**
and their kernels hash stable as well, to reduce recompilation.

### CODE REVIEW

**Comment by jwnys** [comment]
Does this avoid recompiling when the potential function is made time-dependent?

File: netket/operator/_kinetic.py [hunk 1]
```diff
@@ -102,5 +104,11 @@ def _expect_kernel(
     def _pack_arguments(self) -> PyTree:
```
Comment (gcarleo) [inline netket/operator/_kinetic.py:104]: this is really ugly, I hope at some point we can modify it... in fact what is called and stored and mass is not the mass, it's the inverse of the mass...!

**Review by gcarleo — APPROVED** [review]
Looks good to me but I would ask @gpescia to have a look too

**Comment by PhilipVinc** [comment]
This might break code that defines custom/new continuous operators which @jwnys or @gpescia might have, but it's a good improvement and there is no easy way to avoid breaking past code so I will merge anyway.

In case this happens, know that you should define the `_attrs` property and it should return a tuple with everything to be used to compare for equality and hash, as documented internally (those docstrings are not surfaced anywhere as its all quite internal, but will be one day).

**Comment by PhilipVinc** [comment]
> Does this avoid recompiling when the potential function is made time dependent?

It does not, though with this it should be possible to build something that won't recompile.

**Pull request merged**